### PR TITLE
Submodule refactor

### PR DESCRIPTION
**Pull request by gareth-j**
This completes most of the work from #19. Some functions and filenames still need to changed and simplified though. Test coverage could also be improved.

### CODE REVIEW

File: doc/source/api/index_devapi.rst [hunk 1]
```diff
@@ -35,11 +39,165 @@ These classes are used for the processing of data by the ``ObsSurface`` processi
 :class:`~openghg.modules.THAMESBARRIER`
     For processing data from the Thames Barrier measurement sites
 
-* Datasource
+Datasource
+^^^^^^^^^^
 
 The Datasource is the smallest data provider within the OpenGHG topology. A Datasource represents a data provider such as an instrument
```
Comment (rt17603) [inline doc/source/api/index_devapi.rst:45]: What do you mean by "data provider" here - class that provides data?

Comment (gareth-j) [reply doc/source/api/index_devapi.rst:45]: Yeah, that needs changing really as it's confusing. Maybe it should be "store of data" but I think the whole bit could be rewritten.

**Comment by gareth-j** [comment]
Let's rename `process` to `standardise`.

File: notebooks/tutorials/beginner_workflow/footprint_workflow.ipynb [hunk 2]
```diff
@@ -25,7 +25,7 @@
     "import os\n",
     "import matplotlib.pyplot as plt\n",
     "from openghg.localclient import get_obs_surface\n",
-    "from openghg.modules import Emissions, FOOTPRINTS, ObsSurface\n",
+    "from openghg.modules import Emissions, Footprints, ObsSurface\n",
```
Comment (rt17603) [inline notebooks/tutorials/beginner_workflow/footprint_workflow.ipynb:28]: Does the import statement here need to be updated to import from somewhere other than `openghg.modules`? Is it `openghg.store` now?

File: notebooks/tutorials/beginner_workflow/footprint_workflow.ipynb [hunk 3]
```diff
@@ -25,7 +25,7 @@
...
     "from openghg.processing import footprints_data_merge, search\n",
```
Comment (rt17603) [inline notebooks/tutorials/beginner_workflow/footprint_workflow.ipynb:29]: Similiarly so these functions need to be imported from somewhere else now?

**Comment by rt17603** [comment]
Had a quick scan over the files - looks good overall to me. I made a note where the import statements need to be updated. This will apply to the following files (at least) in `notebooks/tutorials/beginner_workflow`:
 - workflow.ipynb
 - footprint_workflow.ipynb
 - footprint_co2_workflow.ipynb

File: notebooks/tutorials/beginner_workflow/footprint_workflow.ipynb (hunk 2 above)
Comment (gareth-j) [reply notebooks/tutorials/beginner_workflow/footprint_workflow.ipynb:28]: Good catch, I'll change them now.

File: notebooks/tutorials/beginner_workflow/footprint_workflow.ipynb (hunk 3 above)
Comment (gareth-j) [reply notebooks/tutorials/beginner_workflow/footprint_workflow.ipynb:29]: I'll change these as well.

**Comment by gareth-j** [comment]
> Had a quick scan over the files - looks good overall to me. I made a note where the import statements need to be updated. This will apply to the following files (at least) in `notebooks/tutorials/beginner_workflow`:
> 
>     * workflow.ipynb
> 
>     * footprint_workflow.ipynb
> 
>     * footprint_co2_workflow.ipynb
> 
> 
> I can apply these updates if you want but didn't want to interfere if you had more ongoing commits to make to this PR.

I'll make the changes now, we also need to get the extra tutorials included in the documentation.

**Comment by gareth-j** [comment]
OK the changes to the other tutorials are done.

**Pull request merged**